packages/@uppy/companion/src/server/controllers/url.js

```diff
@@ -43,15 +43,24 @@ const meta = (req, res) => {
  * @param {object} res expressJS response object
  */
 const get = (req, res) => {
-  logger.debug('URL file import handler running', null, req.id)
-  const { debug } = req.companion.options
-  if (!validateURL(req.body.url, debug)) {
-    logger.debug('Invalid request body detected. Exiting url import handler.', null, req.id)
-    return res.status(400).json({ error: 'Invalid request body' })
-  }
+  (async () => {
+    try {
+      logger.debug('URL file import handler running', null, req.id)
+      const { debug } = req.companion.options
+      if (!validateURL(req.body.url, debug)) {
+        logger.debug('Invalid request body detected. Exiting url import handler.', null, req.id)
+        res.status(400).json({ error: 'Invalid request body' })
+        return
+      }
+
+      // If we already have size, no need to get it again.
+      // See https://github.com/transloadit/uppy/issues/3034
+      let size = req.body && req.body.size
+      if (!size) {
+        const urlMeta = await reqUtil.getURLMeta(req.body.url, !debug)
+        size = urlMeta.size
+      }
```

Comment:
Can `req.body` be nullish? If so, line 60 should throw (`req.body.url`); in any case, line 58 can be simplified:
```suggestion
      let { size } = req.body
      if (!size) {
        const urlMeta = await reqUtil.getURLMeta(req.body.url, !debug)
        size = urlMeta.size
      }
```

Reply:
Right! thanks. I added the check for req.body because I was confused by the method name `get`. I was scared that HTTP GET requests wouldn't always have a req.body so I added the check. But apparently I only fixed it half way. Anyhow it turns out that the method named `get` is actually a POST, so it should always have a `body` (?)